### PR TITLE
Improve the 'invalid name' error message in modtool

### PR DESCRIPTION
**Pull request by fgerlits**
... and fix some validations which checked a prefix instead of the whole name.  https://github.com/gnuradio/gnuradio/issues/3616

### CODE REVIEW

**Comment by mormj** [comment]
@fgerlits - thank you for the contribution!  Love the consolidation of common code to one place and cleaning up the regex match.  I will give this a test shortly.

Would you mind signing your commit since we adhere to the DCO for licensing?  Just a `git commit -s --amend` and then a force push should do the trick

**Comment by mormj** [comment]
Getting an error with the import of validate name:
```
josh@josh-m15:/share/gnuradio/gr310/src/gnuradio/build$ gr_modtool newmod tmp
ERROR:root:cannot import name 'validate_name' from 'gnuradio.modtool.cli.base' (/share/gnuradio/gr310/lib/python3/dist-packages/gnuradio/modtool/cli/base.py)
```

**Comment by mormj** [comment]
Looks like the relative import for validate_name needs to go somewhere else so that it grabs it from core/base.py instead of trying to grab it from cli/base.py

**Comment by fgerlits** [comment]
> Looks like the relative import for validate_name needs to go somewhere else so that it grabs it from core/base.py instead of trying to grab it from cli/base.py

Yes, you're right.  I've fixed it in https://github.com/gnuradio/gnuradio/commit/ece93c6397ae85d2408b00713eb84f9182be43e6.

**Review by mormj — APPROVED** [review]
This looks good to me!  Would like to get @swap-nil7 's eyes on this as well.

File: gr-utils/modtool/tests/test_modtool.py [hunk 1]
```diff
@@ -239,12 +239,19 @@ def test_rename(self):
...
         # Missing 2 arguments blockname, new_name
         self.assertRaises(ModToolException, ModToolRename(**test_dict).run)
+
         test_dict['blockname'] = 'square_ff'
         # Missing argument new_name
         self.assertRaises(ModToolException, ModToolRename(**test_dict).run)
+
         test_dict['new_name'] = '//#'
         # Invalid new block name!
+        self.assertRaisesRegex(ModToolException, "Invalid new block name '//#': names can only contain letters, numbers and underscores", ModToolRename(**test_dict).run)
```
Comment (swap-nil7) [inline gr-utils/modtool/tests/test_modtool.py:249]: Please reduce the length of this line. :)

**Review by swap-nil7 — APPROVED** [review]
Looks good. Thanks for your contribution.

**Review by mormj — APPROVED** [review]

**Pull request merged**